icu4c/source/test/intltest/numbertest_api.cpp

```diff
@@ -2699,6 +2700,31 @@ void NumberFormatterApiTest::unitGender() {
     assertEquals("getGender for a genderless language", "", fn.getGender(status));
 }
 
+void NumberFormatterApiTest::unitNotConvertible() {
+    IcuTestErrorCode status(*this, "unitNotConvertible");
+    const double randomNumber = 1234;
+
+    NumberFormatter::with()
+        .unit(MeasureUnit::forIdentifier("meter-and-liter", status))
```

Comment:
I think this test is not good enough, from this test, it cannot tell which line trigger the error.
Is the error happen in MeasureUnit::forIdentifier() or the error happen in formatDouble()?
If the error is expected to happen in MeasureUnit::forIdentifier(), then the .locale("en_US")
        .formatDouble(randomNumber, status); part should be removed. If the error should happen only in formatDouble() then we should assert the status is not error after the calling of  MeasureUnit::forIdentifier("meter-and-liter", status)